composer/datasets/in_context_learning_evaluation.py

```diff
@@ -1288,15 +1293,27 @@ def __init__(
             'generation_kwargs': {
                 'pad_token_id': self.pad_tok_id,
                 'num_beams': 1,  # single beam
-                'num_return_sequences': generations_per_sample,
                 'do_sample': True,
+                'temperature': 0.2, # good default for code
                 'use_cache': True,
                 'eos_token_id': self.tokenizer.eos_token_id
-            }
+            },
+            'sample_id': [],
+            'pass_at_k': list(pass_at_k),
+            'generations_per_sample': generations_per_sample,
+            'dataset_size': dataset_size,
         }
         if 'generation_kwargs' in kwargs:
             self.update_generation_kwargs(kwargs['generation_kwargs'])
 
+    def repeat_dataset(self, dataset: HFDataset, repetitions: int) -> HFDataset:
+        def repeated_dataset():
+            for i, sample in enumerate(dataset):
```

Comment:
This is very similar to how `bigcode-evaluation-harness` implements a repeated_dataset. Not sure if you got it from them, but their code to do this can be found [here](https://github.com/bigcode-project/bigcode-evaluation-harness/blob/00967d12093ef614de7bdad0772aed8e4118f1fd/bigcode_eval/utils.py#L123).

Reply:
I figured this one independently, but thanks for the pointer!